ksql-engine/src/test/java/io/confluent/ksql/integration/SecureIntegrationTest.java

```diff
@@ -0,0 +1,157 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.ksql.integration;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.test.TestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.confluent.ksql.KsqlEngine;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.testutils.EmbeddedSingleNodeKafkaCluster;
+import io.confluent.ksql.testutils.secure.ClientTrustStore;
+import io.confluent.ksql.testutils.secure.SecureKafkaHelper;
+import io.confluent.ksql.util.KafkaTopicClientImpl;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.OrderDataProvider;
+import io.confluent.ksql.util.PersistentQueryMetadata;
+import io.confluent.ksql.util.QueryMetadata;
+import io.confluent.ksql.util.TopicProducer;
+
+import static io.confluent.ksql.testutils.EmbeddedSingleNodeKafkaCluster.VALID_USER1;
+
+/**
+ * Tests covering integration with secured components, e.g. secure Kafka cluster.
+ */
+public class SecureIntegrationTest {
+
+  @ClassRule
+  public static final EmbeddedSingleNodeKafkaCluster SECURE_CLUSTER =
+      EmbeddedSingleNodeKafkaCluster.newBuilder()
+          .withSaslSslListenersOnly()
+          .build();
+
+  private static final String INPUT_TOPIC = "orders_topic";
+  private static final String INPUT_STREAM = "ORDERS";
+
+  private QueryId queryId;
+  private KsqlEngine ksqlEngine;
+  private TopicProducer topicProducer;
+  private KafkaTopicClientImpl topicClient;
+
+  @Before
+  public void before() {
+    topicProducer = new TopicProducer(SECURE_CLUSTER);
+  }
+
+  @After
+  public void after() {
+    if (queryId != null) {
+      ksqlEngine.terminateQuery(queryId, true);
+    }
+    ksqlEngine.close();
+    topicClient.close();
+  }
+
+  @Test
+  public void shouldRunQueryOnSecureKafkaCluster() throws Exception {
+    // Given:
+    final Map<String, Object> configs = getBaseKsqlConfig();
+
+    // Additional Properties required for KSQL to talk to secure cluster:
+    configs.put("security.protocol", "SASL_SSL");
+    configs.put("sasl.mechanism", "PLAIN");
+    configs.put("sasl.jaas.config", SecureKafkaHelper.buildJaasConfig(VALID_USER1));
+
+    givenTestSetupWithConfig(configs);
+
+    // Then:
+    assertCanRunKsqlQuery();
+  }
+
+  private void givenTestSetupWithConfig(final Map<String, Object> ksqlConfigs) throws Exception {
+    // Additional Properties required for KSQL to talk to test secure cluster,
```

Comment:
nit: most of this could go in the `@Before`?

Reply:
Not really, more testing coming, which will need different config.

I've moved the produce call into `setUp` though.